ProjFS.Mac/PrjFSKext/PrjFSKext/VirtualizationRoots.cpp

```diff
@@ -58,12 +59,16 @@ kern_return_t VirtualizationRoots_Cleanup()
 
 VirtualizationRoot* VirtualizationRoots_FindForVnode(vnode_t vnode)
 {
+    ProfileSample functionSample(Probe_VirtualizationRoot_Find);
```

Comment:
Is the compiler smart enough to avoid the calls to the constructor and destructor when `PRJFS_PERFORMANCE_TRACING_ENABLE` is not defined?  Or would we need to use macros (to avoid creating `ProfileSample `s in the first place)

Reply:
The `ProfileSample` member functions are all defined as `inline`, and as their bodies evaluate to nothing, and there isn't a vtable, they *should* disappear completely on optimised, non-profiling builds. With compiler optimisation, few things are ever guaranteed of course…